internal/pkg/manifest/svc.go

```diff
@@ -148,6 +148,30 @@ func (c *Count) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the ScalingConfigOrPercentage
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (r *ScalingConfigOrPercentage) UnmarshalYAML(value *yaml.Node) error {
```

Comment:
We probably need to implement a transformer for this struct so that environment override applies nicely: https://github.com/aws/copilot-cli/blob/c5fb60645ccef3fab823a36a551d49c927716f2b/internal/pkg/manifest/transform.go.

I'd also be fine with implementing the transformer in an upcoming PR. But just to explain a little why we need it:

Right now, likely both of these manifests ⬇️  
```yaml
count:
  range: 1-3
  cpu_percentage: 30    

environments:
  test:
    count:
      range: 1-3
      cpu_percentage:
        value: 40   
        cooldown:
           in: 3m 
```
and 
```yaml
count:
  range: 1-3
  cpu_percentage:
        value: 40   
        cooldown:
           in: 3m   

environments:
  test:
    count:
      range: 1-3
      cpu_percentage: 30
```

would result in a manifest struct that both has:
```go
AdvancedCount.CPU.Value = 30
AdvancedCount.CPU.ScalingConfig.Value = 40
AdvancedCount.CPU.ScalingConfig.Cooldown.ScaleInCooldown = 3m
```
That is, both `AdvancedCount.CPU.Value` and `AdvancedCount.CPU.ScalingConfig` are non-empty, while at most one of them should be non-empty. It'd be then ambiguous which - `AdvancedCount.CPU.Value` or `AdvancedCount.CPU.ScalingConfig` - contains the value that the user truly want to use (in the first case, `AdvancedCount.CPU.ScalingConfig`  would be the true answer; in the second case, `AdvancedCount.CPU.Value`; but we wouldn't know if they are both non-empty)

Let me know if this make sense!

Reply:
This was the next step for what I was going to do with the feature, thanks for letting me know something I'll need to take into consideration 👍